backend/app/api/src/helpers/AuthenticatedStructures.ts

```diff
@@ -125,44 +127,151 @@ export class AuthenticatedStructures {
+                organizationId: {
+                    sign: 'IN',
+                    value: Array.from(organizationMap.keys())
+                }
+            });
+
+            for(const organizationRegistrationPeriod of result) {
+                const organizationId = organizationRegistrationPeriod.organizationId;
+                const organization = organizationMap.get(organizationId);
+                if(organization) {
+                    organizationsWithRegistrationPeriods.push([organization, organizationRegistrationPeriod]);
+                }
+            }
+        }
+        //#endregion
+
+        //#region check if private data access
+        const organizationsWithoutPrivateDataAccess: [Organization, OrganizationRegistrationPeriod][] = [];
+        const organizationsWithPrivateDataAccess: [Organization, OrganizationRegistrationPeriod][] = [];
+
+        for(const item of organizationsWithRegistrationPeriods) {
+            const organization = item[0];
+
+            // This method is async but it should not call the database because the organization is already fetched.
+            const canAccess = await Context.optionalAuth?.canAccessPrivateOrganizationData(organization);
+            if(canAccess) {
+                // If at least two organizations can be accessesed, than all organizations can be accessed,
+                // but code can change so it is better to keep checking all organizations.
+                organizationsWithPrivateDataAccess.push(item);
+            } else {
+                organizationsWithoutPrivateDataAccess.push(item);
+            }
+        }
+        //#endregion
+        
+        //#region get periods
+        const allPeriodIds = periodIdOrganizationsMap.keys();
+        const allPeriods = await RegistrationPeriod.getByIDs(...allPeriodIds);
+        const periodMap = new Map<string, RegistrationPeriod>(allPeriods.map(p => [p.id, p]));
+        //#endregion
+
+        //#region create organization structs
+        const results: OrganizationStruct[] = [];
+
+        for(const [organization, organizationRegistrationPeriod] of organizationsWithoutPrivateDataAccess) {
+            const registrationPeriod = periodMap.get(organization.periodId);
+            if(!registrationPeriod) {
+                console.error('Registration period is undefined.')
+                continue;
+            }
 
-        if (await Context.optionalAuth?.canAccessPrivateOrganizationData(organization)) {
-            const webshops = await Webshop.where({ organizationId: organization.id }, { select: Webshop.selectColumnsWithout(undefined, "products", "categories")})
-            const webshopStructures: WebshopPreview[] = [] 
+            const result = OrganizationStruct.create({
+                ...organization.getBaseStructure(),
+                period: await this.organizationRegistrationPeriod(organizationRegistrationPeriod, [registrationPeriod])
+            });
+
+            results.push(result);
```

Comment:
De volgorde van de output is niet stabiel. De resultaten moeten in exact dezelfde volgorde worden teruggegeven als de input organizations